aip/0192.md

```diff
@@ -0,0 +1,97 @@
+---
+aip:
+  id: 192
+  state: reviewing
+  created: 2019-07-25
+permalink: /192
+redirect_from:
+  - /0192
+---
+
+# Documentation
+
+Documentation is one of the most critical aspects of API design. Users of your
+API are unable to dig into the implementation to understand the API better;
+often, the API surface definition and its corresponding documentation will be
+the only things a user has. Therefore, it is important that documentation be as
+clear, complete, and unambiguous as possible.
+
+## Guidance
+
+In APIs defined in protocol buffers, public comments **must** be included over
+every component (service, method, message, field, enum, and enum value) using
+the protocol buffers comment format. This is important even in cases where the
+comment is terse and uninteresting, as numerous tools read these comments and
+use them.
+
+Services, in particular, **should** have descriptive comments that explain what
+the service is and what users are able to do with it.
+
+**Note:** Many readers will not be native English speakers. Comments **should**
+avoid jargon, slang, complex metaphors, pop culture references, or anything
+else that will not easily translate. Additionally, many readers will have
+different backgrounds and viewpoints; if writing examples involving people,
+prefer people who are non-controversial and no longer alive.
+
+### Style
+
+Comments **should** generally be in gramatically correct American English.
+However, the first sentence of each comment **should** omit the subject:
+
+```proto
+// Creates a book on the given shelf.
+rpc CreateBook(CreateBookRequest) returns (Book) {
+  option (google.api.http) = {
+    post: "/v1/{parent=shelves/*}/books"
+    body: "book"
+  };
+}
+```
+
+### Descriptions
+
+Descriptions of messages and fields **should** be brief but complete. Sometimes
+comments are necessarily perfunctory because there is nothing else to be said.
+Before jumping to that conclusion, however, consider whether some of the
+following questions are relevant:
+
+- What is it?
+- What does it do if it succeeds? What does it do if it fails? What can cause
+  it to fail, and how?
+- Is it idempotent?
+- What are the units? (Examples: meters, degrees, pixels.)
+- What range of values does it accept? Is the range inclusive or exclusive?
```

Comment:
Possibly highlight what this means for strings.

Reply:
I am unsure what you mean here.

Reply:
For strings, we need to know what length it can be, what characters are allowed etc. That's effectively a special case of "range of values" but an important one.

Reply:
That makes sense. Done.